pkg/skaffold/sync/sync.go

```diff
@@ -86,11 +99,56 @@ func NewItem(a *latest.Artifact, e filemon.Events, builds []build.Artifact, inse
 		return nil, nil
 	}
 
-	return &Item{
-		Image:  tag,
-		Copy:   toCopy,
-		Delete: toDelete,
-	}, nil
+	return &Item{Image: tag, Copy: toCopy, Delete: toDelete}, nil
+}
+
+func inferredSyncItem(a *latest.Artifact, e filemon.Events, builds []build.Artifact, provider DestinationProvider) (*Item, error) {
+	// deleted files are no longer contained in the syncMap, so we need to rebuild
+	if len(e.Deleted) > 0 {
+		return nil, nil
+	}
+
+	tag := latestTag(a.ImageName, builds)
```

Comment:
nit: combine with the next line

```golang
if tag: latestTag(a.ImageName, builds); tag == "" {
  ...
}
```

Reply:
It's also used in the return value in line 151, so that's not possible :(

Reply:
ah, didn't catch that.